### ISSUE TITLE
Fix nested tree relations

### ISSUE THREAD

**Issue by Tronix117** [Wikodit/js-data-jsonapi-light#20]
When we have a structure like this:
```
- category A
--- category A.1
----- category A.1.a
--- category A.2
- category B
```
with `parent`, `children` relationships, it produces some errors when JS-Data handles the serialized datas.

**Comment by Tronix117** [comment]
Follow-up: Js-Data v3 seems to not handle this case correctly, maybe it needs to be assigned to JSData.

See: js-data/js-data#463

**Comment by Tronix117** [comment]
Fixed on jsdata/js-data#463
